### PR TITLE
add generated getter/setter type with thisValue

### PR DESCRIPTION
**Pull request by dylan-conway**

### CODE REVIEW

File: src/bun.js/scripts/generate-classes.ts [hunk 1]
```diff
@@ -609,7 +609,9 @@ function renderDecls(symbolName, typeName, proto) {
         `extern "C" JSC::EncodedJSValue ${symbolName(
           typeName,
           proto[name].getter || proto[name].accessor.getter,
-        )}(void* ptr, JSC::JSGlobalObject* lexicalGlobalObject);`,
+        )}(void* ptr,${
+          proto[name].this!! ? " JSC::EncodedJSValue thisValue, " : ""
```
Comment (Jarred-Sumner) [inline src/bun.js/scripts/generate-classes.ts]: Should it be 

```
!!proto[name].this ? 
```

File: src/bun.js/scripts/generate-classes.ts [hunk 2]
```diff
@@ -1317,27 +1324,39 @@ function generateZig(
     }
 
     [...Object.values(proto)].forEach(
-      ({ getter, setter, accessor, fn, cache, DOMJIT }) => {
+      ({ getter, setter, accessor, fn, this: thisValue, cache, DOMJIT }) => {
```
Comment (Jarred-Sumner) [inline src/bun.js/scripts/generate-classes.ts]: Maybe default it to false 

```
this: thisValue = false 
```

**Review by Jarred-Sumner — APPROVED** [review]
LGTM some comments but it's fine

File: src/bun.js/scripts/generate-classes.ts (hunk 1 above)
Comment (dylan-conway) [reply src/bun.js/scripts/generate-classes.ts]: 👍

File: src/bun.js/scripts/generate-classes.ts (hunk 2 above)
Comment (dylan-conway) [reply src/bun.js/scripts/generate-classes.ts]: 👍

**Pull request merged**